doc/design/dynamic_rnn.md

```diff
@@ -0,0 +1,126 @@
+# Implementation Doc: Dynamic RNN
+
+## A glance of Dynamic RNN
+
+A common neural network structure called recurrent neural network(`RNN` for short), which there is a directed circle in the neural network model. RNN can use a internal memory to process arbitrary sequences of inputs.
```

Comment:
a internal memory --> an internal memory
```process arbitrary sequences of inputs``` is not clear enough. Maybe you mean:
RNN can use an internal memory to process sequences with variable lengths.

Reply:
This sentence is quoted from https://en.wikipedia.org/wiki/Recurrent_neural_network